pkgs/by-name/au/autobase/package.nix

```diff
@@ -27,7 +33,60 @@ buildNpmPackage (finalAttrs: {
     cp ${./package-lock.json} ./package-lock.json
   '';
 
-  passthru.updateScript = nix-update-script { };
+  passthru = {
+    updateScriptSrc = gitUpdater {
+      rev-prefix = "v";
+    };
+    updateScriptVendor = writeShellApplication {
+      name = "update-autobase-lockfile-npmDepsHash";
+      runtimeInputs = [
+        nix
+        nodejs
+        prefetch-npm-deps
+        sd
+      ];
+      text = ''
+        export UPDATE_NIX_ATTR_PATH="''${UPDATE_NIX_ATTR_PATH:-autobase}"
+
+        oldversion="''${UPDATE_NIX_OLD_VERSION-}"
+        newversion="$(nix-instantiate . --eval --strict -A "$UPDATE_NIX_ATTR_PATH.version" | cut -d'"' -f2)"
+
+        if [ "$oldversion" == "$newversion" ]; then
+          echo "No new version."
+          exit 0
+        fi
+
+        # For NPM to behave nice
+        HOME="$(mktemp -d)"
```

Comment:
Could this be changed to `writableTmpDirAsHomeHook` in `runtimeInputs` instead? Not sure.

Reply:
`updateScript`s don't run as a derivation with a strong sandbox or phases or anything AFAIU, they're kinda just executables that get invoked with some special environment variables set, and within your writable copy of Nixpkgs. I doubt that adding hook derivations works… Could maybe try to `source` the underlying script file, and then run `writableTmpDirAsHome`, but *is that really much better than a quick `mktemp -d`,,,*